src/daemon/emitter.cc

```diff
@@ -595,12 +534,16 @@ void Emitter::DoPoll(const base::WorkerPool& pool,
     }
   };
 
-  do {
+  auto conf = this->conf();
+  const auto poll_interval = conf->emitter().poll_interval();
+
+  while (!pool.WaitUntilShutdown(std::chrono::seconds(poll_interval))) {
```

Comment:
When switching `do - while` to `while` emitter may wait too long for getting remotes from coordinator, isn't it?
If `poll_interval` equals 30 mins, for example.